src/main/java9/org/jboss/modules/JDKSpecific.java

```diff
@@ -188,6 +192,36 @@ static void addInternalPackages(final List<String> list) {
         // none in Java 9+
     }
 
+    static <T> Iterable<T> findServices(final Class<T> serviceType, final Predicate<Class<?>> filter, final ClassLoader classLoader) {
+       final Iterator<ServiceLoader.Provider<T>> delegate = ServiceLoader.load(serviceType, classLoader).stream().iterator();
+       return new Iterable<T>() {
+            public Iterator<T> iterator() {
+                return new Iterator<T>() {
+                    T next = null;
+
+                    public boolean hasNext() {
+                        ServiceLoader.Provider<T> next;
```

Comment:
Same problem as above - if `this.next` is not `null` then `true` should be returned.

Reply:
Fixed